builtin/logical/pki/path_acme_new_account.go

```diff
@@ -304,14 +311,24 @@ func (b *backend) acmeNewAccountSearchHandler(acmeCtx *acmeContext, r *logical.R
 	return nil, fmt.Errorf("An account with this key does not exist: %w", ErrAccountDoesNotExist)
 }
 
-func (b *backend) acmeNewAccountCreateHandler(acmeCtx *acmeContext, r *logical.Request, fields *framework.FieldData, userCtx *jwsCtx, data map[string]interface{}, contact []string, termsOfServiceAgreed bool) (*logical.Response, error) {
+func (b *backend) acmeNewAccountCreateHandler(acmeCtx *acmeContext, userCtx *jwsCtx, contact []string, termsOfServiceAgreed bool) (*logical.Response, error) {
 	if userCtx.Existing {
 		return nil, fmt.Errorf("cannot submit to newAccount with 'kid': %w", ErrMalformed)
 	}
 
 	// If the account already exists, return the existing one.
-	if b.acmeState.DoesAccountExist(acmeCtx, userCtx.Kid) {
-		return b.acmeNewAccountSearchHandler(acmeCtx, r, fields, userCtx, data)
+	thumbprint, err := userCtx.GetKeyThumbprint()
+	if err != nil {
+		return nil, fmt.Errorf("failed generating thumbprint for key: %w", err)
+	}
+
+	accountByKey, err := b.acmeState.LoadAccountByKey(acmeCtx, thumbprint)
+	if err != nil {
```

Comment:
This needs to be `if err != nil && !errors.Is(err, ErrAccountDoesNotExist)`, right, with the comment about why?

Reply:
I don't believe so, `LoadAccountByKey` now traps the `ErrAccountDoesNotExist` coming from `LoadAccount` and returns a nil. nil response as it used too.